karabo_data/reader.py

```diff
@@ -1058,6 +1062,52 @@ def RunDirectory(path):
 RunHandler = RunDirectory
 
 
+def open_run(proposal, run, proc=True):
+    """Access data from a specified run in the EuXFEL files on Maxwell.
+
+    Parameters
+    ----------
+    proposal: str, int
+        A proposal number, such as 2012, '2012', 'p002012', or a path such as
+        '/gpfs/exfel/exp/SPB/201701/p002012'.
+    run: str, int
+        A run number such as 243, '243' or 'r0243'.
+    proc: bool
+        If True (default), combine data from the raw and 'proc' (processed)
+        folders for the specified run. If False, access only the raw data.
+    """
+    if isinstance(proposal, int):
+        proposal = 'p{:06d}'.format(proposal)
+    elif ('/' not in proposal) and not proposal.startswith('p'):
+        proposal = 'p' + proposal.rjust(6, '0')
+
+    prop_dir = find_proposal(proposal)
```

Comment:
I wonder if we should add an additional test in case the user has not read access to the proposal, because right now the error message is `no .h5 files found in ...` which is not helpful, maybe something like:

```python
if not os.access(prop_dir, os.R_OK):
    raise Exception(f'cannot access {prop_dir}: permission denied')
```

Reply:
Rather than adding an explicit test, I've switched from `glob()` to `os.scandir()` (which is like listdir but a bit fancier). `glob()` suppresses errors encountered while it's scanning, whereas using the lower level `os` functions means a standard PermissionError should be raised.